File: src/export/microsoft_ads_exporter.py
import pandas as pd

MATCH_TYPE_MAP = {
    "Exact match": "Exact",
    "Exact": "Exact",
    "Phrase match": "Phrase",
    "Phrase": "Phrase",
    "Broad match": "Broad",
    "Broad": "Broad",
}


def export_microsoft_ads_csv(df: pd.DataFrame) -> str:
    out = pd.DataFrame()
    out["Type"] = "Keyword"
    out["Status"] = "Active"
    out["Campaign"] = df["campaign"].values
    out["Ad Group"] = df["ad_group"].values
    out["Keyword"] = df["keyword"].values
    out["Match Type"] = (
        df["match_type"].map(MATCH_TYPE_MAP).fillna("Broad").values
    )
    out["Bid"] = df["target_cpc"].map(lambda x: f"{x:.2f}").values
    return out.to_csv(index=False)

File: src/lib/bid-engine/microsoft-ads-exporter.ts
import Papa from "papaparse";
import type { KeywordRow } from "./types";

const MATCH_TYPE_MAP: Record<string, string> = {
  "Exact match": "Exact",
  Exact: "Exact",
  "Phrase match": "Phrase",
  Phrase: "Phrase",
  "Broad match": "Broad",
  Broad: "Broad",
};

export function exportMicrosoftAdsCsv(rows: KeywordRow[]): string {
  const data = rows.map((row) => ({
    Type: "Keyword",
    Status: "Active",
    Campaign: row.campaign,
    "Ad Group": row.ad_group,
    Keyword: row.keyword,
    "Match Type": MATCH_TYPE_MAP[row.match_type] || "Broad",
    Bid: (row.target_cpc ?? 0).toFixed(2),
  }));

  return Papa.unparse(data);
}
